config/initializers/action_pack.rb

```diff
@@ -13,7 +13,11 @@ def send_file_with_content_length(path, options = {})
   alias_method_chain :send_file, :content_length
 
   def send_data_with_content_length(data, options = {})
-    headers.merge!('Content-Length' => data.length.to_s) if data.respond_to?(:length)
+    if RUBY_VERSION > '1.9'
```

Comment:
given that canvas will not boot on ruby < 1.9, you don't need a conditional here - just change the one line and not worry about it.